File: 2019 Kakao Winter Internship CodingTest/2.js
function solution(s) {
    let recoveredTuple = [];
    let tupleSet = s.split("{").filter(r=>r!=='');
    tupleSet = tupleSet.map(i => {
        return i.replace('}','').replace('}','').split(',').filter(r=>r!=='');
    });


    for(let i=1;i<=tupleSet.length;i++) {
        tupleSet.find(set => set.length===i).forEach(e=>{
            if(recoveredTuple.indexOf(Number(e))<0) {
                recoveredTuple.push(Number(e));
                return;
            }
        });

    }

    console.log(tupleSet);
    console.log(recoveredTuple);
}

solution("{{2},{2,1},{2,1,3},{2,1,3,4}}");
solution("{{1,2,3},{2,1},{1,2,4,3},{2}}");

File: 2019 Kakao Winter Internship CodingTest/2/2.py
def solution(s):
    s = s[1:-1].split('},')
    tuple_set = [ [] for _ in range(len(s))]
    answer = []
    for i in range(len(s)):
        s[i] = s[i][1:]
        if s[i][-1] == '}':
            s[i] = s[i][:-1]
        tuple_set[i] = list(map(int, s[i].split(',')))

    tuple_set.sort(key=len)
    
    print(tuple_set)

    for i in tuple_set:
        for j in i:
            if answer.count(j) == 0:
                answer.append(j)
    print(answer)
solution('{{1,2,3},{2,1},{1,2,4,3},{2}}')
